fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchReferenceTest.java

```diff
@@ -34,22 +36,27 @@ protected void setTenant() throws Exception {
         FHIRRequestContext.get().setTenantId("reference");
     }
 
+    @BeforeClass
+    public void createReference() throws FHIRException {
+        String originalRequestUri = "https://example.com/Patient/123";
+        FHIRRequestContext context = FHIRRequestContext.get();
+        context.setOriginalRequestUri(originalRequestUri);
+    }
+
     @Test
     public void testSearchReference_Reference_relative() throws Exception {
         // Reference by id really only works when the system knows which resource type(s)
         // can be referenced from a given element.
-        // TODO does this work if you define the extension in a StructureDefinition
-        // and declare the allowed types?
-//        assertSearchReturnsSavedResource("Reference-relative", "123");
-
         assertSearchReturnsSavedResource("Reference-relative", "Patient/123");
+        assertSearchReturnsSavedResource("Reference-relative", "123");
 
-        // TODO if this matched the hostname that the test was running on, would it work?
-//        assertSearchReturnsSavedResource("Reference-relative", "https://example.com/Patient/123");
+        assertSearchReturnsSavedResource("Reference-relative", "https://example.com/Patient/123");
 
         assertSearchReturnsSavedResource("Reference-relative:Patient", "123");
-        assertSearchDoesntReturnSavedResource("Reference-relative:Basic", "123");
 
+        // Originally the following test was set to not return a value, as we are now using 123 as a valid value, it
+        // does return the expected resource
+        assertSearchReturnsSavedResource("Reference-relative:Basic", "123");
```

Comment:
I don't think this is right.  The value of the extension extracted by Reference-relative is `Patient/123`, whereas here we are explicitly saying we'd like a Basic resource.

Reply:
that may be my fault... I'll double check this afternoon

Reply:
fixed, I had to pass down the modified resource target